src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala

```diff
@@ -149,102 +183,141 @@ class AppDefinitionTest extends MarathonSpec with Matchers {
     }
 
     val idError = "path contains invalid characters (allowed: lowercase letters, digits, hyphens, \".\", \"..\")"
-    val app = AppDefinition(id = "a b".toRootPath)
+    var app = AppDefinition(id = "a b".toRootPath)
+    validateJsonSchema(app, false)
+    shouldViolate(app, "id", idError)
+
+    app = app.copy(id = "a#$%^&*b".toRootPath)
+    validateJsonSchema(app, false)
+    shouldViolate(app, "id", idError)
+
+    app = app.copy(id = "-dash-disallowed-at-start".toRootPath)
+    validateJsonSchema(app, false)
+    shouldViolate(app, "id", idError)
+
+    app = app.copy(id = "dash-disallowed-at-end-".toRootPath)
+    validateJsonSchema(app, false)
+    shouldViolate(app, "id", idError)
 
+    app = app.copy(id = "uppercaseLettersNoGood".toRootPath)
+    validateJsonSchema(app, false)
     shouldViolate(app, "id", idError)
-    shouldViolate(app.copy(id = "a#$%^&*b".toRootPath), "id", idError)
-    shouldViolate(app.copy(id = "-dash-disallowed-at-start".toRootPath), "id", idError)
-    shouldViolate(app.copy(id = "dash-disallowed-at-end-".toRootPath), "id", idError)
-    shouldViolate(app.copy(id = "uppercaseLettersNoGood".toRootPath), "id", idError)
-    shouldNotViolate(app.copy(id = "ab".toRootPath), "id", idError)
 
+    val app2 = AppDefinition(id = "test".toPath, instances = -3, ports = Seq(9000, 8080, 9000))
     shouldViolate(
-      AppDefinition(id = "test".toPath, instances = -3, ports = Seq(9000, 8080, 9000)),
+      app2,
       "ports",
       "Elements must be unique"
     )
+    validateJsonSchema(app2, false)
 
+    val app3 = AppDefinition(id = "test".toPath, ports = Seq(0, 0, 8080))
     shouldNotViolate(
-      AppDefinition(id = "test".toPath, ports = Seq(0, 0, 8080)),
+      app3,
       "ports",
       "Elements must be unique"
     )
+    validateJsonSchema(app3, false)
 
     val correct = AppDefinition(id = "test".toPath)
 
+    var appX = correct.copy(executor = "//cmd")
```

Comment:
The tests use app1, app2, etc, and since this gets reused several times, I just named it appX.